### PR TITLE
fix(core):  missing array modifier for scalar fields

### LINKED ISSUE #346: JSON Schema export does not include abstract types
# Bug Report 🐛

JSON Schema generation currently filters out abstract types; breaking export of models that contain arrays of abstract types.

https://github.com/accordproject/concerto/blob/master/packages/concerto-tools/lib/codegen/fromcto/jsonschema/jsonschemavisitor.js#L166

## Expected Behavior

We need to include abstract types, so that reference to these types are valid.

## Current Behavior

Exported JSON Schema is invalid if it includes abstract types, and those abstract types are referenced in arrays/properties.

## Possible Solution
Include the abstract types.

## Steps to Reproduce

1.
2.
3.
4.

### Desktop
 - OS: [e.g. macOS]
 - Browser: [e.g. Chrome, Safari]
 - Version: [e.g. 0.22.15]

### LINKED ISSUE #745: JSON Schema output is not working for Scalar Arrays
# Bug Report 🐛 The JSON Schema generated from a concerto model is incorrect for `scalar array` types.

Consider the below model:
```
     scalar SSN extends String default="000-00-0000" regex=/\d{3}-\d{2}-\d{4}/
      concept Product identified by sku
      {
          o Double d1
          o Double[] d2
          o SSN[] ssn
          o SSN ssn2
      }
```
Generating a JSONSchema for the above works fine for all except the `SSN[]` type.
```json
    "catalog@1.0.0.Product": {
      "title": "Product",
      "description": "An instance of catalog@1.0.0.Product",
      "type": "object",
      "properties": {
        "$class": {
          "type": "string",
          "default": "catalog@1.0.0.Product",
          "pattern": "^catalog@1\\.0\\.0\\.Product$",
          "description": "The class identifier for catalog@1.0.0.Product"
        },
        "d1": {
          "type": "number"
        },
        "d2": {
          "type": "array",
          "items": {
            "type": "number"
          }
        },
        "ssn": {
          "default": "000-00-0000",
          "type": "string",
          "pattern": "\\d{3}-\\d{2}-\\d{4}"
        },
        "ssn2": {
          "default": "000-00-0000",
          "type": "string",
          "pattern": "\\d{3}-\\d{2}-\\d{4}"
        }
      },
      "required": [
        "$class",
        "d1",
        "d2",
        "ssn",
        "ssn2"
      ]
    },
```
Both `SSN` and `SSN[]` types are generated as `string` and `string array` in `SSN[]` case.

## Expected Behavior

The `Scalar Array` types should be generated as `array` types correctly in the JSON Schema output.

## Current Behavior

`Sclar array` is generated as `scalar` type without the array modifier.

## Context (Environment)

    "@accordproject/concerto-codegen": "^3.16.2",
    "@accordproject/concerto-core": "^3.13.0",
    "@apidevtools/json-schema-ref-parser": "^11.1.0",

### PR DESCRIPTION
**Pull request by KrishnaPG**
# Closes #745 

### Changes

- added array modifier for scalar field unboxing condition

### Related Issues
- Issue #745

### Author Checklist
- [x] Ensure you provide a [DCO sign-off](https://github.com/probot/dco#how-it-works) for your commits using the `--signoff` option of git commit.
- [x] Vital features and changes captured in unit and/or integration tests
- [x] Commits messages follow [AP format](https://github.com/accordproject/techdocs/blob/master/DEVELOPERS.md#commit-message-format)
- [x] Extend the documentation, if necessary
- [x] Merging to `main` from `fork:branchname`

### CODE REVIEW

**Review by mttrbrts — COMMENTED** [review]
Thanks for the fix, GK. Are you able to add a test too, please?

**Comment by KrishnaPG** [comment]
@mttrbrts Tried adding few tests. Please review.

**Comment by mttrbrts** [comment]
> @mttrbrts Tried adding few tests. Please review.

Looking good, thanks! A couple of linting failures. But I'm happy to merge once those are fixed

**Comment by KrishnaPG** [comment]
Fixed the lint issues. Thanks.

**Review by mttrbrts — APPROVED** [review]
Thank you for your first PR @KrishnaPG  🎉

We should get a release out in the next few days.

File: packages/concerto-core/test/introspect/scalars.js [hunk 1]
```diff
@@ -171,5 +171,21 @@ describe('Scalars', () => {
             p.getScalarField().getType().should.equal('DateTime');
         });
 
+        it('should handle arrays correctly', () => {
+            const p = new Field(mockClassDeclaration, {
+                $class: `${MetaModelNamespace}.StringProperty`,
+                name: "property",
```
Comment (mttrbrts) [inline packages/concerto-core/test/introspect/scalars.js:177]:
```suggestion
                name: 'property',
```

**Pull request merged**